### DISCUSSION TITLE
How to Set Up Advanced Security for Public Repositories?

### DISCUSSION THREAD

**Discussion by davewichers** [community/community#116434] [Code Security] [answered]
### Topic

Setting Up GitHub Advanced Security for Public Repositories

### Question

Per: [About licenses for GitHub Advanced Security](https://docs.github.com/en/billing/managing-billing-for-github-advanced-security/about-billing-for-github-advanced-security#about-licenses-for-github-advanced-security), it says: "If you want to use GitHub Advanced Security features on any repository apart from a public repository on GitHub.com, ...". 

However, this page, nor any other page I can find, explains how to enable GitHub Advanced Security on a public repository. Can someone either explain how, or point me to the proper page that does explain how? In my case, this is a personal project, not one in an organization.

**Comment by aashah** [comment]
Advanced Security is a paid set of products, so unfortunately that may not be available.

However, many of those security products are still offered for free public repositories. For example, Secret Scanning is available -- https://docs.github.com/en/code-security/secret-scanning/configuring-secret-scanning-for-your-repositories

Most of the security product settings should appear on the same page, based on their availability. Lemme know if you were looking for a specific security product.

**Reply by davewichers** [reply]
Both employees of GitHub, via emails, and the above GitHub page says you can use GitHub Advanced Security on public / open source repos. But they don't explain HOW to enable it. That's what I'm looking for.

**Reply by aashah** [reply]
If you have paid for advanced security, it will be available from your repo settings > `Code security and analysis` > `GitHub Advanced Security`.

<img width="1151" alt="Screenshot 2024-04-01 at 19 14 58" src="https://github.com/community/community/assets/1176726/9f59e730-7f46-477e-98c4-3f6577836c23">

It is **not** available if your org or business has not purchased licenses.

The page you linked above is specifically about how Advanced Security is billed, not it's availability. What the documentation was trying to say is that contributions made to public repositories do not _consume_ licenses, only contributions to private repositories. Advanced Security is still only available by purchasing licenses. You can see the subtext in the screenshot above say the same-ish thing.

**Comment by shaikhmubin02** [comment]
Enable Two-Factor Authentication (2FA): Enable 2FA for all collaborators to add an extra layer of security.

**Comment by davewichers** [comment, marked as answer]
I guess I should have been more clear in my question. I'm specifically interested in piloting the new **Autofix** feature. However, per: https://github.com/orgs/community/discussions/111094 (**Code scanning autofix: Preview Feedback and Resources**) - it says: "Fix suggestions are available on private repositories with a working code scanning configuration." And 2 weeks ago there was also the comment: "We do not plan on shipping it to open source at the moment." so I guess as of today, you can't enable Advanced Security Autofix on public repos. You have to either pay for it or use a 30-day trial on an organization which has 1 or more private repos.

GH also confirmed this: "Autofix is currently only available for private repositories owned by GHAS customers. We’re tracking the request to extend it to open source developers [here](https://github.com/orgs/community/discussions/111094#discussioncomment-8861383)."

**Reply by aashah** [reply]
Got it. Yeah that would have been helpful to specify...Like I mentioned in the other comment, some security products and their features have become available to open source repositories. Advanced Security however, is always a paid feature.
